docs/errors/E0383.md

```diff
@@ -0,0 +1,15 @@
+# E0383: variable assignment to self is no-op
+
+The following is a no-op and will warn:
```

Comment:
Nit: "will warn" is misleading. When run with a JavaScript engine, there is no warning. I would say just "The following is a no-op:".

Reply:
Would a better description be "useless variable assignment" instead of referring to it as a "no-op"?